pages/1_orders_and_revenue.py

```diff
@@ -16,51 +16,180 @@
-
-## KPI Metrics which need to be updated.
+# KPI Metrics
 with st.container():
     col1, col2, col3, col4 = st.columns(4)
 
+    total_revenue = data['total_amount'].sum()
+    number_of_orders = data['header_id'].nunique()
+    number_of_customers = data['customer_id'].nunique()
+    min_created_at = data['created_at'].min()
+    max_created_at = data['created_at'].max()
+    new_customers = data[(data['customer_created_at'] >= min_created_at) & (data['customer_created_at'] <= max_created_at)].shape[0]
+
     with col1:
-        st.metric(label="**Total Revenue**", value="TBD")
+        st.metric(label="**Total Revenue**", value=f"${total_revenue:,.2f}")
 
     with col2:
-        st.metric(label="**Number of Orders**", value="TBD")
+        st.metric(label="**Number of Orders**", value=number_of_orders)
 
     with col3:
-        st.metric(label="**Number of Customers**", value="TBD")
+        st.metric(label="**Number of Customers in Period**", value=number_of_customers)
 
     with col4:
-        st.metric(label="**New Customers**", value="TBD")
+        st.metric(label="**New Customers**", value=new_customers)
+
+st.divider()
+
+# Time series charts
+data['month'] = data['created_at'].dt.to_period('M').dt.to_timestamp()
 
-# Time series charts need to be built out
 with st.container():
     row1_col1, row1_col2 = st.columns(2)
     row2_col1, row2_col2 = st.columns(2)
 
     with row1_col1:
         st.markdown("**Total Revenue Over Time**")
+        revenue_over_time = data.groupby('month')['total_amount'].sum().reset_index()
+        fig = px.bar(revenue_over_time, x='month', y='total_amount')
+        fig.update_yaxes(tickprefix='$', tickformat='~s', title_text='Revenue', range=[0, revenue_over_time['total_amount'].max() * 1.1])
+        fig.update_xaxes(title_text='Month', tickformat='%b %Y')
+        st.plotly_chart(fig, use_container_width=True)
 
     with row1_col2:
         st.markdown("**Total Orders Over Time**")
+        orders_over_time = data.groupby('month')['header_id'].nunique().reset_index()
+        fig = px.bar(orders_over_time, x='month', y='header_id')
+        fig.update_yaxes(title_text='Orders', range=[0, orders_over_time['header_id'].max() * 1.1])
+        fig.update_xaxes(title_text='Month', tickformat='%b %Y')
+        st.plotly_chart(fig, use_container_width=True)
 
     with row2_col1:
         st.markdown("**Percent of Successful Payments Over Time**")
+        successful_payments = data[data['header_status'] == 'completed'].groupby('month')['payment_id'].nunique()
+        total_payments = data.groupby('month')['payment_id'].nunique()
+        percent_successful = (successful_payments / total_payments * 100).fillna(0).reset_index()
+        fig = px.bar(percent_successful, x='month', y='payment_id')
+        fig.update_yaxes(range=[0, 100], title_text='Percentage (%)')
+        fig.update_xaxes(title_text='Month', tickformat='%b %Y')
+        st.plotly_chart(fig, use_container_width=True)
```

Comment:
Let's make the update here that was discussed in the last review to focus on product revenue

Reply:
updated

Reply:
@fivetran-reneeli this looks good, but the colors are a bit confusing. What information are the different colors providing for each product? If we are doing a vertical bar chart in this way I imagine the colors could all be the same, right?

Reply:
Sure, updated